composer/trainer/ddp.py

```diff
@@ -79,6 +93,16 @@ def ddp_sync_context(state: State, is_final_microbatch: bool, sync_strategy: Uni
 
 
 def prepare_ddp_module(module: Model, find_unused_parameters: bool) -> Model:
```

Comment:
Thoughts about removing this function and just embedding it in the trainer? It's called only in one place, and the trainer already guards calls to this function to ensure that dist is configured and initialized.

Reply:
Edit: Since we can't make `ddp.py` private I'm going to leave it here since it is "ddp logic" and it's nice to keep it in a separate file from the trainer which is already a huge file - lmk if you think otherwise though and I can move it